src/core/ext/filters/channel_idle/channel_idle_filter.cc

```diff
@@ -277,9 +278,10 @@ void ChannelIdleFilter::StartIdleTimer() {
 
 void ChannelIdleFilter::CloseChannel() {
   auto* op = grpc_make_transport_op(nullptr);
-  op->disconnect_with_error = grpc_error_set_int(
-      GRPC_ERROR_CREATE("enter idle"),
-      StatusIntProperty::ChannelConnectivityState, GRPC_CHANNEL_IDLE);
+  op->disconnect_with_error =
+      GRPC_ERROR_BUILDER(kUnknown, "enter idle")
+          .Set(StatusIntProperty::ChannelConnectivityState, GRPC_CHANNEL_IDLE)
```

Comment:
Since the list of properties is known, I'd suggest:
`.WithChannelConnectivityState(GRPC_CHANNEL_IDLE)`

Reply:
Is this to reduce the code length?

Reply:
also to improve readability... there's a lot to read here and most of it's noise that doesn't help illustrate the intent of the code